### PR TITLE
Skipped Privacy Modules

### PR DESCRIPTION
**Pull request by AntoxaAntoxic**
### 🔧 Type of changes
- [ ] new bid adapter
- [ ] bid adapter update
- [ ] new feature
- [ ] new analytics adapter
- [ ] new module
- [ ] module update
- [x] bugfix
- [ ] documentation
- [ ] configuration
- [ ] dependency update
- [ ] tech debt (test coverage, refactorings, etc.)

### ✨ What's the context?
What's the context for the changes?

### 🧠 Rationale behind the change
Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🔎 New Bid Adapter Checklist
- [ ] verify email contact works
- [ ] NO fully dynamic hostnames
- [ ] geographic host parameters are NOT required
- [ ] direct use of HTTP is prohibited - *implement an existing Bidder interface that will do all the job*
- [ ] if the ORTB is just forwarded to the endpoint, use the generic adapter - *define the new adapter as the alias of the generic adapter*
- [ ] cover an adapter configuration with an integration test

### 🧪 Test plan
How do you know the changes are safe to ship to production?

### 🏎 Quality check
- [ ] Are your changes following [our code style guidelines](https://github.com/prebid/prebid-server-java/blob/master/docs/developers/code-style.md)?
- [ ] Are there any breaking changes in your code?
- [ ] Does your test coverage exceed 90%?
- [ ] Are there any erroneous console logs, debuggers or leftover code in your changes?

### CODE REVIEW

File: src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java [hunk 1]
```diff
@@ -34,6 +41,7 @@ public ActivityInfrastructureDebug(String accountId,
         this.accountId = accountId;
         this.traceLevel = traceLevel;
         this.traceLog = new ArrayList<>();
+        this.skippedModules = new HashSet<>();
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java:44]: EnumSet

File: src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java [hunk 2]
```diff
@@ -56,6 +64,16 @@ public void emitActivityInvocationDefaultResult(boolean defaultResult) {
     }
 
     public void emitProcessedRule(Rule rule, Rule.Result result) {
+        if (rule instanceof AbstainPrivacyModule module) {
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java:67]: Rename to `SkippedPrivacyModule`

File: src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java [hunk 3]
```diff
@@ -56,6 +64,16 @@ public void emitActivityInvocationDefaultResult(boolean defaultResult) {
...
+            skippedModules.add(module.skippedModule());
+        } else if (rule instanceof AndRule andRule) {
+            CollectionUtils.emptyIfNull(andRule.rules()).stream()
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java:70]: Can't be `null`

File: src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java [hunk 4]
```diff
@@ -56,6 +64,16 @@ public void emitActivityInvocationDefaultResult(boolean defaultResult) {
...
+                    .filter(AbstainPrivacyModule.class::isInstance)
+                    .findFirst()
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java:72]: We need to add all entries, there is a possibility for different modules be there

File: src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java [hunk 5]
```diff
@@ -56,6 +64,16 @@ public void emitActivityInvocationDefaultResult(boolean defaultResult) {
...
+                    .map(AbstainPrivacyModule.class::cast)
+                    .ifPresent(module -> skippedModules.add(module.skippedModule()));
+        }
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java:75]: Also, extract this part to separate method `collectSkippedPrivacyModules`

File: src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java [hunk 6]
```diff
@@ -15,14 +19,17 @@
...
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 
 public class ActivityInfrastructureDebug {
 
     private final String accountId;
     private final TraceLevel traceLevel;
     private final List<ExtTraceActivityInfrastructure> traceLog;
+    private final Set<PrivacyModuleQualifier> skippedModules;
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebug.java:32]: Rename to `skippedPrivacyModules`
Also, rename other occurrences

File: src/test/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebugTest.java [hunk 7]
```diff
@@ -156,6 +161,52 @@ public void emitProcessedRuleShouldReturnExpectedResultIfTraceLevelIsVerbose() {
...
+        verifyNoMoreInteractions(metrics);
+    }
+
+    @Test
+    public void emitProcessedRuleShouldLogSkippedModuleWhenAndRuleHasAbstainModule() {
+        // given
+        final ActivityInfrastructureDebug debug = debug(TraceLevel.verbose);
+
+        // when
+        debug.emitProcessedRule(
```
Comment (CTMBNara) [inline src/test/java/org/prebid/server/activity/infrastructure/debug/ActivityInfrastructureDebugTest.java:192]: Also, add test to check when `traceLevel` is `null`

**Review by CTMBNara — APPROVED** [review]
